zookeeper-server/src/main/java/org/apache/zookeeper/ClientCnxn.java

```diff
@@ -1510,6 +1511,11 @@ public void close() throws IOException {
      * the server. Thus, getXid() must be public.
      */
     synchronized public int getXid() {
+        // xid values of -4, -2, and -1 are special, see SendThread.readResponse
```

Comment:
What about introducing constants for these special values ? (and replace in readResponse)

Reply:
Where do protocol level constants like that usually get put?

Reply:
I am thinking about having constants only _locally_ to this file, this way it will be easier to understand the code and the patch wouldn't be so intrusive.
In the (hopefully near) future we will separate client size code from server side code so introducing new shared constants would be overkilling.

I don't fell strong about having such constants, it is only a thought/suggestion.

cc @anmolnar @lvfangmin @hanm

Reply:
I'm inclined to keep the patch minimal since grepping for those identifiers wouldn't tell you where those packets get sent.  Also, the numerical values are useful to see here.  I don't feel strongly about it though.

Reply:
no problem from my side, let's keep the patch simple